walletnode/services/userdataprocess/task.go

```diff
@@ -0,0 +1,391 @@
+			}
+			aggregate[hex.EncodeToString(hashvalue)] = append(aggregate[hex.EncodeToString(hashvalue)],i)
+		}
+	}
+
+	if count < task.config.MinimalNodeConfirmSuccess {
+		// If there is not enough reponse from supernodes
+		return userdata.ProcessResult{
+			ResponseCode: userdata.ErrorNotEnoughSupernodeResponse,
+			Detail:       userdata.Description[userdata.ErrorNotEnoughSupernodeResponse],
+		}, nil
+	}
+
+	max := 0
+	var finalHashKey string
+	for key, element := range aggregate {
+		if max < len(element) {
+			max = len(element)
+			finalHashKey = key
+		}
+	}
+
+	if len(aggregate[finalHashKey]) < task.config.MinimalNodeConfirmSuccess {
+		return userdata.ProcessResult{
+			ResponseCode: userdata.ErrorNotEnoughSupernodeConfirm,
+			Detail:       userdata.Description[userdata.ErrorNotEnoughSupernodeConfirm],
+		}, nil
+	}
+
+	// There is enough supernodes verified our request, so we return that response
+	if len(aggregate[finalHashKey]) > 0 && nodes[(aggregate[finalHashKey])[0]] != nil {
+		result := *(nodes[(aggregate[finalHashKey])[0]].Result)
+		return result, nil
+	} */
+
+	return userdata.ProcessResult{}, errors.Errorf("failed to Aggregate Result")
+}
+
+// meshNodes establishes communication between supernodes.
+func (task *Task) meshNodes(ctx context.Context, nodes node.List, primaryIndex int) (node.List, error) {
+	var meshNodes node.List
+
+	primary := nodes[primaryIndex]
+	if err := primary.Connect(ctx, task.config.connectTimeout); err != nil {
+		return nil, err
+	}
+	if err := primary.Session(ctx, true); err != nil {
+		return nil, err
+	}
+	primary.SetPrimary(true)
+
+	if len(nodes) == 1 {
+		// If the number of nodes only have 1 node, we use this primary node and return directly
+		meshNodes.Add(primary)
+		return meshNodes, nil
+	}
+
+	nextConnCtx, nextConnCancel := context.WithCancel(ctx)
+	defer nextConnCancel()
+
+	var secondaries node.List
+	go func() {
+		for i, node := range nodes {
+			node := node
+
+			if i == primaryIndex {
+				continue
+			}
+
+			select {
+			case <-nextConnCtx.Done():
+				return
+			case <-time.After(task.config.connectToNextNodeDelay):
+				go func() {
+					defer errors.Recover(log.Fatal)
+
+					if err := node.Connect(ctx, task.config.connectTimeout); err != nil {
+						return
+					}
+					if err := node.Session(ctx, false); err != nil {
+						return
+					}
+					secondaries.Add(node)
+
+					if err := node.ConnectTo(ctx, primary.PastelID(), primary.SessID()); err != nil {
+						return
+					}
+					log.WithContext(ctx).Debugf("Seconary %q connected to primary", node)
+				}()
+			}
+		}
+	}()
+
+	acceptCtx, acceptCancel := context.WithTimeout(ctx, task.config.acceptNodesTimeout)
+	defer acceptCancel()
+
+	accepted, err := primary.AcceptedNodes(acceptCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	meshNodes.Add(primary)
+	for _, pastelID := range accepted {
+		log.WithContext(ctx).Debugf("Primary accepted %q secondary node", pastelID)
+
+		node := secondaries.FindByPastelID(pastelID)
```

Comment:
Potential race issue here. See the fix on reg-art.

Reply:
fixed, thank you